packages/radio-player/src/search-bar/search-bar.ts

```diff
@@ -0,0 +1,248 @@
+import {
+  LitElement,
+  html,
+  css,
+  customElement,
+  property,
+  CSSResult,
+  TemplateResult,
+  PropertyValues,
+} from 'lit-element';
+
+import magnifyingGlassIcon from './assets/img/magnifying-glass';
+import disclosureTriangleIcon from './assets/img/disclosure';
+import clearResultsIcon from './assets/img/clear-results';
+
+@customElement('search-bar')
+export default class SearchBar extends LitElement {
+  @property({ type: Boolean }) isOpen = false;
+
+  @property({ type: Boolean }) showsDisclosure = false;
+
+  @property({ type: String }) searchTerm = '';
+
+  @property({ type: Array }) quickSearches: string[] = [];
+
+  render(): TemplateResult {
+    return html`
+      <div
+        class="
+          container
+          ${this.isOpen ? 'is-open' : ''}
+          ${this.showsDisclosure ? 'shows-disclosure' : ''}"
+      >
+        <div class="search-bar">
+          <div class="magnifier-container endcap">
+            ${magnifyingGlassIcon}
+          </div>
+          <input
+            type="text"
+            class="search-input"
+            placeholder="Search"
+            value=${this.searchTerm}
+            @keyup=${this.inputChanged}
+          />
+          <div
+            class="clear-search-container endcap ${this.searchTerm === '' ? '' : 'is-searching'}"
+          >
+            <button @click=${this.clearSearch}>
+              ${clearResultsIcon}
+            </button>
+          </div>
+          <div class="disclosure-container endcap">
+            <button @click=${this.toggleDisclosure}>
+              ${disclosureTriangleIcon}
+            </button>
+          </div>
+        </div>
+        <div class="quick-search">
+          <quick-search
+            .quickSearches=${this.quickSearches}
+            @searchTermSelected=${this.doQuickSearch}
+          >
+          </quick-search>
+        </div>
+      </div>
+    `;
+  }
+
+  private clearSearch(): void {
+    this.searchTerm = '';
+    this.emitSearchClearedEvent();
+  }
+
+  private inputChanged(e: KeyboardEvent): void {
+    this.emitInputChangeEvent();
+    if (e.key === 'Enter') {
+      this.emitEnterKeyPressedEvent();
+    }
+  }
+
+  private emitInputChangeEvent(): void {
+    const value = this.searchInput && this.searchInput.value;
+    const event = new CustomEvent('inputchange', {
+      detail: { value },
+      bubbles: true,
+      composed: true,
+    });
+    this.dispatchEvent(event);
+  }
+
+  private emitEnterKeyPressedEvent(): void {
+    const value = this.searchInput && this.searchInput.value;
```

Comment:
**Suggestion**: change this line to a ternary that uses either `value` or `''` for the assignment, then on line 94 simply pass the variable as a shorthand property `{ value }`